### PR TITLE
Organize and update dependencies

### PR DESCRIPTION
**Pull request by Tapchicoma**
- Update kotlin to `1.2.30` version
- Update Gradle to `4.6` version
- Update default Ktlint version in extension to `0.19.0` version

Add a little dependency organization.

### CODE REVIEW

File: plugin/buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+object Versions {
+    val kotlin = "1.2.30"
+    val ktlintPlugin = "3.1.0"
+    val gradlePublishPlugin = "0.9.10"
+    val androidPlugin = "3.0.1"
+    val kotlinNativePlugin = "0.6.1"
+    val semver = "1.0.0"
+    val gradleWrapper = "4.6"
+}
```
Comment (JLLeitschuh) [inline plugin/buildSrc/src/main/kotlin/Dependencies.kt]: ❓ Some redundancy here? Any way to reduce it?

Comment (Tapchicoma) [reply plugin/buildSrc/src/main/kotlin/Dependencies.kt]: I tried with `ln -s` as *plugin* is a separate Gradle project, but got `Redeclaration: Versions` error on compile. Can't figure out any other solution.

Comment (JLLeitschuh) [reply plugin/buildSrc/src/main/kotlin/Dependencies.kt]: Use a different name for the two classes? Call one `BuildSrcVersions` and the other something else?

**Comment by Tapchicoma** [comment]
@JLLeitschuh check updated commit. Now samples dependencies depend on plugin dependencies versions.

File: buildSrc/src/main/kotlin/PluginDependencies.kt [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../../plugin/buildSrc/src/main/kotlin/PluginDependencies.kt
```
Comment (JLLeitschuh) [inline buildSrc/src/main/kotlin/PluginDependencies.kt]: Is this a symlink or something??

Comment (Tapchicoma) [reply buildSrc/src/main/kotlin/PluginDependencies.kt]: yes, symlink. Without it `samples` project doesn't see `PluginVersions` class.

**Pull request merged**